modules/libcom/src/osi/epicsStdio.h

```diff
@@ -183,6 +184,140 @@ LIBCOM_API int epicsStdCall epicsStdoutVPrintf(
 LIBCOM_API int epicsStdCall epicsStdoutPuts(const char *str);
 LIBCOM_API int epicsStdCall epicsStdoutPutchar(int c);
 
+/**
+ * \brief Open file from memory or disk.
+ * \param pathname Path of file to open.
+ * \param mode Open Mode.  For a memory file only "r" or "rb" are accepted.
+ * \returns NULL on error and sets `errno`
+ *
+ * If `pathname` is prefixed with `emem://`, then an entry
+ * previously passed to epicsFMount() may be returned.
+ * Otherwise, the call is forwarded to the OS fopen().
+ *
+ * \note `emem://` files can only be opened for read-only access,
+ *       and may not have an associated file descriptor.
+ *
+ * \note `emem://` is not available on vxWorks.
+ *
+ * \since UNRELEASED
+ */
+LIBCOM_API
+FILE* epicsFOpen(const char *pathname, const char *mode);
+
+/**
+ * \brief Open file memory file.
+ * \param pathname Path of file to open.
+ * \param mode Open Mode.  Only "r" or "rb" are accepted.
+ * \returns NULL on error and sets `errno`
+ *
+ * `pathname` must not be prefixed with `emem://`.
+ *
+ * \since UNRELEASED
+ */
+LIBCOM_API
+FILE* epicsMemOpen(const char *pathname, const char *mode);
+
+/** bit mask of possible compression types */
+#define EPICS_MEM_FILE_ALGO_MASK (0xff)
+
+/** Content array is raw bytes (uncompressed) */
+#define EPICS_MEM_FILE_RAW (0x1)
+
+/** Content array is zlib compressed */
+#define EPICS_MEM_FILE_ZLIB (0x2)
```

Comment:
Choice of algorithm is an enumuration using the first 8 bits.  Thus the `*ALGO_MASK` macro, and composition with the `*NOCOPY` bit (`0x100`).  So I guess the question is really whether the `flags` member should be overloaded in this way vs. individual member fields.

I'm slightly inclined towards the combined `flags` as it seems somewhat easier to accommodate future additions.

Although, it might be a good idea to include a "canary" field at the end to ensure that callers zero initialize the `epicsIMF` struct.

Reply:
Even if you overload `flags`, algorithms can still be an enum which makes it clear they are not flags. Although, I don't see what you gain with the overloaded approach. If you split the 64-bit `flags` field into 32-bit `flags` and 32-bit `algorithm`, the struct will be no bigger and you will still have 32 flags; if you run out of those, it means the API is getting too complicated.

Reply:
I've split `epicsIMF::flags` up into an enum and a C bit field.